src/pages/PersonPage/PersonPage.tsx

```diff
@@ -1,86 +1,169 @@
-    ];
+  const fetchCouncilmemberRoles = useCallback(async () => {
+    const roleService = new RoleService(firebaseConfig);
+    const roles = await roleService.getCouncilMemberRolesByPersonId(id);
+    return Promise.resolve(roles);
+  }, [id, firebaseConfig]);
+  const { state: councilmemberRolesDataState } = useFetchData<Role[]>(
+    { ...initialFetchDataState },
+    fetchCouncilmemberRoles
+  );
 
+  const fetchSeatPicture = useCallback(async () => {
+    const fileService = new FileService(firebaseConfig);
+    const { networkService } = fileService;
     if (
-      councilMemberRoles.length > 0 &&
-      councilMemberRoles[0].seat &&
-      councilMemberRoles[0].seat.image_ref
+      councilmemberRolesDataState.data &&
+      councilmemberRolesDataState.data.length > 0 &&
+      councilmemberRolesDataState.data[0].seat &&
+      councilmemberRolesDataState.data[0].seat.image_ref
     ) {
-      const seatImage = await fileService.getFileById(councilMemberRoles[0].seat.image_ref);
-      pictureSrcPromises.push(networkService.getDownloadUrl(seatImage.uri));
-    } else {
-      pictureSrcPromises.push(Promise.resolve(undefined));
+      const seatImage = await fileService.getFileById(
+        councilmemberRolesDataState.data[0].seat.image_ref
+      );
+      const seatImageSrc = await networkService.getDownloadUrl(seatImage.uri);
+      return Promise.resolve(seatImageSrc);
     }
+    return Promise.resolve(null);
+  }, [councilmemberRolesDataState.data, firebaseConfig]);
+  const { state: seatPictureDataState, dispatch: seatPictureDataDispatch } = useFetchData<
+    string | null
+  >({ ...initialFetchDataState, hasFetchRequest: false }, fetchSeatPicture);
+  useEffect(() => {
+    if (councilmemberRolesDataState.data) {
+      // fetch the seat picture when the council member roles are fetched
+      seatPictureDataDispatch({ type: FetchDataActionType.FETCH });
+    }
+  }, [councilmemberRolesDataState.data, seatPictureDataDispatch]);
 
-    const [personPictureSrc, seatPictureSrc] = await Promise.all(pictureSrcPromises);
+  const fetchRoles = useCallback(async () => {
+    const roleService = new RoleService(firebaseConfig);
+    const roles = await roleService.getPopulatedRolesByPersonId(id);
+    return Promise.resolve(roles);
+  }, [id, firebaseConfig]);
+  const { state: rolesDataState } = useFetchData<Role[]>({ ...initialFetchDataState }, fetchRoles);
 
-    const payload: PersonPageData = {
-      person,
-      votes,
-      mattersSponsored,
-      roles,
-      personPictureSrc,
-      seatPictureSrc,
-      councilMemberRoles,
-    };
+  const fetchMatterSponsored = useCallback(async () => {
+    const matterSponsorService = new MatterSponsorService(firebaseConfig);
+    const mattersSponsored = await matterSponsorService.getMattersSponsoredByPersonId(id);
+    return Promise.resolve(mattersSponsored);
+  }, [id, firebaseConfig]);
+  const { state: mattersSponsoredDataState } = useFetchData<MatterSponsor[]>(
```

Comment:
I don't understand JS syntax that well - what exactly is going on in this line? You're setting _something_ on the component's state, but I am not clear what or how.

Reply:
Just renaming `state` to `mattersSponsoredDataState `

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
go to "Assigning to new variable names"

Reply:
`useFetchData` returns an object that has a field `state`. I'm just renaming the variable.